core/bcast/recast.go

```diff
@@ -101,8 +109,25 @@ func (r *Recaster) SlotTicked(ctx context.Context, slot core.Slot) error {
 	}
 	r.mu.Unlock()
 
+	activeVals, err := r.activeValsFunc(ctx)
+	if err != nil {
+		return errors.Wrap(err, "get active validator")
+	}
+
 	for pubkey, tuple := range clonedTuples {
 		ctx := log.WithCtx(ctx, z.Any("duty", tuple.duty))
+
+		ethPk, err := pubkey.ToETH2()
+		if err != nil {
+			log.Error(ctx, "Can't convert pubkey to eth2 format", err)
+			continue
+		}
+
+		if _, found := activeVals[ethPk]; !found {
+			log.Debug(ctx, "Ignoring non-active validator", z.Str("pubkey", pubkey.String()))
```

Comment:
inactive validators is expected, so this is going to be very verbose expeseically for large clusters, can remove I think.

Reply:
yea was about to ask you for an opinion on that debug message, will remove!